### PR TITLE
Create modal dialog - WIP

### PR DESCRIPTION
**Pull request by emanuelmetal**
## Background
To add and edit a User a modal dialog is needed, also for deleting a new one, a confirmation must be done.

## Changes Done

- A modal dialog componen was created
- User form was converted into stateless component
- Action Buttons was separated from Home Page and handles form interactions
- Selected user is stored in redux state
- User Name and Email are validated as required and email format

### CODE REVIEW

File: webapp/app/components/common/modal/MsModal.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  render() {
+    return (
+      <div>
+        <Modal isOpen={this.props.isOpen} toggle={this.props.toggle}>
+          <ModalHeader toggle={this.props.toggle}>{this.props.modalTitle}</ModalHeader>
+          <ModalBody>
+            {this.props.body}
+          </ModalBody>
+          <ModalFooter>
+            <Button color="primary" onClick={this.props.toggle}>{this.props.okButtonLabel}</Button>{' '}
```
Comment (javier-pepe) [inline webapp/app/components/common/modal/MsModal.jsx]: I would avoid computed whitespace as a general rule, preffer to style and add desired spacing via Stylesheet, or inline styles either adding padding to the buttons or the desired margin

File: webapp/app/components/pages/ActionButtons/ActionButtons.jsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
+{/*<Button color="danger" onClick={this.toggle}>{this.props.buttonLabel}</Button>*/}
```
Comment (javier-pepe) [inline webapp/app/components/pages/ActionButtons/ActionButtons.jsx]: I would remove commented out code as a general rule

File: webapp/app/components/pages/ActionButtons/ActionButtons.jsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      modalTitle: 'Delete User',
+      modalYesLabel: 'Delete'
+    }, this.toggle);
+  }
+
+  render() {
+    return (
+      <div>
+        <Button color="primary" onClick={this.toggleAddModal}>Add</Button>{' '}
+        <Button color="info" onClick={this.toggleEditModal} disabled={ this.props.editButtonDisabled }>Edit</Button>{' '}
```
Comment (javier-pepe) [inline webapp/app/components/pages/ActionButtons/ActionButtons.jsx]: I would avoid computed whitespace as a general rule, preffer to style and add desired spacing via Stylesheet, or inline styles either adding padding to the buttons or the desired margin

File: webapp/app/components/common/modal/MsModal.jsx (hunk 1 above)
Comment (emanuelmetal) [reply webapp/app/components/common/modal/MsModal.jsx]: Done, that remained from before your previous comment.

File: webapp/app/components/pages/ActionButtons/ActionButtons.jsx (hunk 2 above)
Comment (emanuelmetal) [reply webapp/app/components/pages/ActionButtons/ActionButtons.jsx]: Done, will update on next push

File: webapp/app/components/pages/home/HomePage.jsx [hunk 4]
```diff
@@ -59,9 +61,11 @@ export class HomePage extends React.Component {
               <h4>Users List</h4>
             </Col>
             <Col md="4">
-              <Button color="primary">Add</Button>{' '}
-              <Button color="info" disabled={ this.state.selectedUser.length === 0 }>Edit</Button>{' '}
-              <Button color="danger" disabled={ this.state.selectedUser.length === 0 }>Delete</Button>{' '}
+              <ActionButtons
+                user={this.state.user}
+                editButtonDisabled={this.state.selectedRow.length === 0}
```
Comment (javier-pepe) [inline webapp/app/components/pages/home/HomePage.jsx]: I would remove props `editButtonDisabled` and `removeButtonDIsabled` and instead implement a prop similar to `isUserSelected` in ActionButtons with its value defined prior in the render method as `const isUserSelected = !this.state.selectedRow.length;` given that condition, ActionButtons could internally disable Edit and Delete (or any new buttons related to user edit actions) this way we avoid evaluating twice for editButtonDIsabled and deleButtonDisabled

File: webapp/app/components/pages/home/HomePage.test.js [hunk 5]
```diff
@@ -6,7 +6,7 @@ function setup(props) {
   return shallow(<HomePage {...props} />);
 }
 
-describe('<HomePage /> component', () => {
+xdescribe('<HomePage /> component', () => {
```
Comment (javier-pepe) [inline webapp/app/components/pages/home/HomePage.test.js]: this test block is disabled

**Review by javier-pepe — CHANGES_REQUESTED** [review]
There are a few improvements required before this PR could be merged

File: webapp/app/components/pages/ActionButtons/ActionButtons.jsx (hunk 3 above)
Comment (emanuelmetal) [reply webapp/app/components/pages/ActionButtons/ActionButtons.jsx]: idem

**Review by javier-pepe — APPROVED** [review]

**Pull request merged**